pints/_mcmc/_population.py

```diff
@@ -0,0 +1,269 @@
+    one whose target distribution we want to sample.
+
+    [1] "On population-based simulation for static inference", Ajay Jasra,
+    David A. Stephens and Christopher C. Holmes, Statistical Computing, 2007.
+    """
+    def __init__(self, x0, sigma0=None):
+        super(PopulationMCMC, self).__init__(x0, sigma0)
+
+        # Set initial state
+        self._running = False
+
+        # Current points, and the logpdf of those points (_not_ the tempered
+        # version!)
+        self._current = None
+        self._current_log_pdf = None
+
+        # Single proposed point
+        self._proposed = None
+
+        # Inner chains
+        self._chains = None
+
+        #
+        # Default settings
+        #
+        self._method = pints.AdaptiveCovarianceMCMC
+
+        # Temperature schedule
+        self._schedule = None
+        self.set_schedule()
+
+        #
+        # Logging
+        #
+        self._have_exchanged = False
+
+    def ask(self):
+        """ See :meth:`SingleChainMCMC.ask()`. """
+        # Initialise on first call
+        if not self._running:
+            self._initialise()
+
+        # Propose new points
+        if self._proposed is None:
+
+            # Select chains to update and exchange/crossover
+            n = len(self._chains)
+            self._i, self._j = np.random.choice(n, 2, replace=False)
+
+            # Propose new point
+            self._proposed = self._chains[self._i].ask()
+
+        # Return proposed point
+        return self._proposed
+
+    def _initialise(self):
+        """
+        Initialises the routine before the first iteration.
+        """
+        if self._running:
+            raise RuntimeError('Already initialised.')
+
+        # Create inner chains
+        n = len(self._schedule)
+        self._chains = [self._method(self._x0, self._sigma0) for i in range(n)]
+
+        # Propose initial point
+        self._current = None
+        self._current_log_pdf = None
+        self._proposed = self._x0
+
+        # Next chain to update
+        self._i = 0
+
+        # Next chain to exchange / crossover with
+        self._j = 1
+
+        # Ask all inner chains for first point (should be x0, so ignore!)
+        for chain in self._chains:
+            chain.ask()
+
+        # Update sampler state
+        self._running = True
+
+    def _log_init(self, logger):
+        """ See :meth:`Loggable._log_init`. """
+        n = len(self._schedule)
+        logger.add_counter('i', max_value=n)
+        logger.add_counter('j', max_value=n)
+        logger.add_string('Ex.', width=3)
+
+    def _log_write(self, logger):
+        """ See :meth:`Loggable._log_write`. """
+        logger.log(self._i)
+        logger.log(self._j)
+        logger.log('yes' if self._have_exchanged else 'no')
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Population MCMC'
+
+    def schedule(self):
```

Comment:
is "schedule" the correct word here, it implies it is changing with time, which I thought it wouldn't?..... "distribution" perhaps?

Reply:
Yeah, agreed. How about temperature_schedule?

Reply:
sounds good, that is much more meaningful

Reply:
Changed it!